include/nbl/video/alloc/SubAllocatedDataBuffer.h

```diff
@@ -228,13 +228,13 @@ class SubAllocatedDataBuffer : public virtual core::IReferenceCounted, protected
             return unallocatedSize;
         }
         //!
-        inline void         multi_free(core::smart_refctd_ptr<IDriverFence>&& fence, DeferredFreeFunctor&& functor) noexcept
+        inline void         multi_free(core::smart_refctd_ptr<IGPUFence>&& fence, DeferredFreeFunctor&& functor) noexcept
         {
             #ifdef _NBL_DEBUG
             std::unique_lock<std::recursive_mutex> tLock(stAccessVerfier,std::try_to_lock_t());
             assert(tLock.owns_lock());
             #endif // _NBL_DEBUG
-            deferredFrees.addEvent(GPUEventWrapper(std::move(fence)),std::forward<DeferredFreeFunctor>(functor));
+            deferredFrees.addEvent(GPUEventWrapper(mDevice, std::move(fence)),std::forward<DeferredFreeFunctor>(functor));
```

Comment:
cant the fence keep the device inside?

Reply:
it does.. but what about this question?

Reply:
also make it work with a host-waitable event

Reply:
fixed by me